src/windows/views/emojis_listview.py

```diff
@@ -52,14 +52,11 @@ def startDrag(self, event):
         """ Override startDrag method to display custom icon """
 
         # Get image of selected item
-        selected_row = self.emojis_model.model.itemFromIndex(self.emojis_model.proxy_model.mapToSource(self.selectionModel().selectedIndexes()[0])).row()
-        icon = self.emojis_model.model.item(selected_row, 0).icon()
+        selected_index = self.selectedIndexes()
 
         # Start drag operation
         drag = QDrag(self)
-        drag.setMimeData(self.emojis_model.proxy_model.mimeData(self.selectionModel().selectedIndexes()))
-        drag.setPixmap(icon.pixmap(QSize(self.drag_item_size, self.drag_item_size)))
-        drag.setHotSpot(QPoint(self.drag_item_size / 2, self.drag_item_size / 2))
+        drag.setMimeData(self.model.mimeData(selected_index))
```

Comment:
All of this stuff was _crazy_ overcomplicated because model data wasn't encapsulated fully in the model items.

Proxy models are just that — they _proxy_ the source model, so if the data is arranged properly there's no reason to have to ever translate references all the way back to the source model. We can retrieve all of the data we need right from the item reference in the (final) proxy model — it will take care of going back through any other proxies in the chain, and ultimately to the source, to access the underlying model data.

(I realized I dropped the pixmap here, I'll push another commit momentarily to restore that.)

Reply:
Fixed, so bumping this comment in the hopes it won't keep getting hidden as "Outdated" since it's not.